runtime/controller/connection.go

```diff
@@ -16,9 +16,19 @@ limitations under the License.
 
 package controller
 
-import "github.com/spf13/pflag"
+import (
+	"fmt"
+	"net/url"
+	"os"
 
-const flagInsecureAllowHTTP = "insecure-allow-http"
+	"github.com/spf13/pflag"
+)
+
+const (
+	flagInsecureAllowHTTP = "insecure-allow-http"
+	EnvVarHTTPProxy       = "HTTP_PROXY"
+	EnvVarHTTPSProxy      = "HTTPS_PROXY"
```

Comment:
I think it's safer to use [httpproxy.FromEnvironment](https://pkg.go.dev/golang.org/x/net/http/httpproxy#FromEnvironment) here as that's the source of truth for Go's runtime.